runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/VmDecompilerInformationController.java

```diff
@@ -196,12 +208,95 @@ private void loadClassBytecode(String name) {
         bytecodeDecompilerView.reloadTextField(decompiledClass);
     }
 
+    private static String lastFile = System.getProperty("user.home");
+
+    private void rewriteClass(String name) {
```

Comment:
What is MVC? Also, this method is way too long. I know that we're not doing all the best practices around here but come on 70 lines? Sorry for being harsh but this code in unreadable.

Reply:
If you make it using MVC you could reuse OkCancelPanel, FileSelectorPanel and maybe more.